src/main/java/graphql/nadel/engine/HydrationInputResolver.java

```diff
@@ -282,12 +327,17 @@ private ExecutionResultNode convertSingleHydrationResultIntoOverallResult(Execut
                                                                               ResultComplexityAggregator resultComplexityAggregator
     ) {
 
-
         Map<String, FieldTransformation> transformationByResultField = queryTransformationResult.getFieldIdToTransformation();
         Map<String, String> typeRenameMappings = queryTransformationResult.getTypeRenameMappings();
+
+        ExecutionResultNode root = rootResultNode.getChildren().get(0);
+        if (hydrationTransformation.getUnderlyingServiceHydration().getSyntheticField() != null && root.getChildren().size() != 0) {
```

Comment:
what happens if we have a synth field but root.getChildren.size() != 0?
We should be explicit and add an Assert here.

Reply:
Could you clarify what you mean since the if statement handles that case

Reply:
we should add an assertion there that root.getChildren.size() ==1  and change the condition to .size() > 0 (to make it more readable)

```
Assert.assertTrue(root.getChildren.size() == 1
```